File: starter-bots/KotlinBot/src/main/kotlin/arctica/service/model/EngineConfigDto.kt
package arctica.service.model

data class EngineConfigDto
    (
    val runnerUrl: String,
    val runnerPort: String,
    val botCount: Int,
    val maxTicks: Int,
    val scoutWorkTime: Int,
    val tickRate: Int,
    val worldLength: Int, // => RegionSize * NumberOfRegionsInMapLength
    val worldArea: Int, // => WorldLength ^ 2
    val regionSize: Int,
    val processTick: Int,

    val baseZoneSize: Int,
    val numberOfRegionsInMapLength: Int,
    val resourceWorldCoverage: Int,
    val populationDecreaseRatio: Double,
    val worldSeed: Int,
    val consumptionRatio: HashMap<String, Float>,
    val scoreRates: HashMap<String, Int>,
    val resourceScoreMultiplier: ResourceScoreMultiplier,
    val unitConsumptionRatio: UnitConsumptionRatio,
    val populationTiers: List<PopulationTier>,
    val startingFood: Int,

    val startingUnits: Int,

    val resourceGenerationConfig: ResourceGenerationConfig,

    val seeds: Seeds,
    val minimumPopulation: Int,

    val resourceImportance: ResourceImportance,
    val minimumUnits: Int
)

data class ResourceGenerationConfig
    (
    val farm: RenewableResourceConfig,
    val wood: NonRenewableResourceConfig,
    val stone: NonRenewableResourceConfig,

    val campfire: ConsumptionResourceConfig
)

open class ResourceConfig
    (
    val ProximityDistance: Int,
    val DistributionZones: List<String>,
    val QuantityRangePerRegion: List<Int>,

    val RewardRange: List<Int>,

    val workTimeRange: List<Int>,
    val maxUnitsRange: List<Int>
)

class ConsumptionResourceConfig
    (
    val resourceConsumption: HashMap<String, List<Int>>,
    ProximityDistance: Int,
    DistributionZones: List<String>,
    QuantityRangePerRegion: List<Int>,
    RewardRange: List<Int>,
    workTimeRange: List<Int>,
    maxUnitsRange: List<Int>
) : ResourceConfig(
    ProximityDistance, DistributionZones, QuantityRangePerRegion, RewardRange, workTimeRange,
    maxUnitsRange
)


class RenewableResourceConfig
    (
    val regenerationRateRange: RegenerationRateRange,
    val AmountRange: List<Int>,
    ProximityDistance: Int,
    DistributionZones: List<String>,
    QuantityRangePerRegion: List<Int>,
    RewardRange: List<Int>,
    workTimeRange: List<Int>,
    maxUnitsRange: List<Int>
) : ResourceConfig(
    ProximityDistance, DistributionZones, QuantityRangePerRegion, RewardRange, workTimeRange,
    maxUnitsRange
)

class NonRenewableResourceConfig
    (
    val amountRange: List<Int>,
    ProximityDistance: Int,
    DistributionZones: List<String>,
    QuantityRangePerRegion: List<Int>,
    RewardRange: List<Int>,
    workTimeRange: List<Int>,
    maxUnitsRange: List<Int>
) : ResourceConfig(
    ProximityDistance, DistributionZones, QuantityRangePerRegion, RewardRange, workTimeRange,
    maxUnitsRange
)

data class RegenerationRateRange
    (
    val tickRange: List<Int>,
    val amountRange: List<Int>
)

data class Seeds
    (
    val PlayerSeeds: List<Int>,
    val MaxSeed: Int,
    val MinSeed: Int
)

data class ResourceScoreMultiplier
    (
    val population: Int,
    val food: Int,
    val wood: Int,
    val stone: Int,
    val gold: Int
)

data class UnitConsumptionRatio
    (
    val food: Double,
    val wood: Double,
    val stone: Double,
    val gold: Double,
    val heat: Double
)

data class UnitActionDuration
    (
    val Farm: Int,
    val Scout: Int,
    val Lumber: Int,
    val Mine: Int
)

data class UnitActionReward
    (
    val farm: Int,
    val lumber: Int,
    val stone: Int,
    val gold: Int
)

data class ResourceImportance
    (
    val food: Double,
    val heat: Double
)

File: starter-bots/NETCoreBot/NETCoreBot/Models/EngineConfigDto.cs
using System.Collections.Generic;
using Domain.Models;
using NETCoreBot.Enums;

namespace NETCoreBot.Models
{
    public class EngineConfigDto
    {
        public string RunnerUrl { get; set; }
        public string RunnerPort { get; set; }
        public int BotCount { get; set; }
        public int MaxTicks { get; set; }
        public int ScoutWorkTime { get; set; }
        public int TickRate { get; set; }
        public int WorldLength => RegionSize * NumberOfRegionsInMapLength;
        public int WorldArea => WorldLength ^ 2;
        public int RegionSize { get; set; }
        public int ProcessTick { get; set; }

        public int BaseZoneSize { get; set; }
        public int NumberOfRegionsInMapLength { get; set; }
        public int ResourceWorldCoverage { get; set; }
        public double PopulationDecreaseRatio { get; set; }
        public int WorldSeed { get; set; }
        public Dictionary<GameObjectType, decimal> ConsumptionRatio { get; set; }
        public Dictionary<GameObjectType, int> ScoreRates { get; set; }
        public ResourceScoreMultiplier ResourceScoreMultiplier { get; set; }
        public UnitConsumptionRatio UnitConsumptionRatio { get; set; }
        public IList<PopulationTier> PopulationTiers { get; set; }
        public int StartingFood { get; set; }

        public int StartingUnits { get; set; }

        public ResourceGenerationConfig ResourceGenerationConfig { get; set; }

        public Seeds Seeds { get; set; }
        public int MinimumPopulation { get; set; }

        public ResourceImportance ResourceImportance { get; set; }
        public int MinimumUnits { get; set; }
    }

    public class ResourceGenerationConfig
    {
        public RenewableResourceConfig Farm { get; set; }
        public NonRenewableResourceConfig Wood { get; set; }
        public NonRenewableResourceConfig Stone { get; set; }

        public ConsumptionResourceConfig Campfire { get; set; }
    }

    public class ResourceConfig
    {
        public int ProximityDistance { get; set; }
        public IList<string> DistributionZones { get; set; }
        public IList<int> QuantityRangePerRegion { get; set; }

        public IList<int> RewardRange { get; set; }

        public IList<int> WorkTimeRange { get; set; }
        public IList<int> MaxUnitsRange { get; set; }
    }

    public class ConsumptionResourceConfig : ResourceConfig
    {
        public Dictionary<ResourceType, IList<int>> ResourceConsumption { get; set; }
    }


    public class RenewableResourceConfig : ResourceConfig
    {
        public RegenerationRateRange RegenerationRateRange { get; set; }
        public IList<int> AmountRange { get; set; }
    }

    public class NonRenewableResourceConfig : ResourceConfig
    {
        public IList<int> AmountRange { get; set; }
    }

    public class RegenerationRateRange
    {
        public IList<int> TickRange { get; set; }
        public IList<int> AmountRange { get; set; }
    }

    public class Seeds
    {
        public List<int> PlayerSeeds { get; set; }
        public int MaxSeed { get; set; }
        public int MinSeed { get; set; }
    }

    public class ResourceScoreMultiplier
    {
        public int Population { get; set; }
        public int Food { get; set; }
        public int Wood { get; set; }
        public int Stone { get; set; }
        public int Gold { get; set; }
    }

    public class UnitConsumptionRatio
    {
        public double Food { get; set; }
        public double Wood { get; set; }
        public double Stone { get; set; }
        public double Gold { get; set; }
        public double Heat { get; set; }
    }

    public class UnitActionDuration
    {
        public int Farm { get; set; }
        public int Scout { get; set; }
        public int Lumber { get; set; }
        public int Mine { get; set; }
    }

    public class UnitActionReward
    {
        public int Farm { get; set; }
        public int Lumber { get; set; }
        public int Stone { get; set; }
        public int Gold { get; set; }
    }

    public class ResourceImportance
    {
        public double Food { get; set; }
        public double Heat { get; set; }
    }
}